web_generator/lib/src/bcd.dart

```diff
@@ -15,12 +15,18 @@ import 'filesystem_api.dart';
 class BrowserCompatData {
   static final Map<String, Set<BCDPropertyStatus>> _eventHandlers = {};
 
+  /// Whether to generate all the bindings regardless of property status.
+  static bool generateAll = false;
```

Comment:
I was trying to see if we could get away without making this static and without too much refactoring, and it appears that maybe it's possible?

Seems we use this property in:
* BCDItem subclasses's `shouldGenerate` getters: those however appear to be used within the context of a method that already checks for generateAll (isEventHandlerSupported and shouldGenerateInterface) so it may be safe to remove the use from these getters?
* the methods where generateAll is used also have access to a translator, so they can fetch the value through it (e.g. `_shouldGenerateMember` can read it from `Translator.instance!.browserCompatibilityData.generateAll`?

Would that work?

Reply:
Good idea! It took a little bit of refactoring, but making `shouldGenerate` a field allows this to work for the first bullet-point and the second bullet point works for translator.dart. `isEventHandlerSupported` doesn't need to check for `generateAll` since `shouldGenerate` handles that anyways.